### PR TITLE
Guest tracing

### PR DESCRIPTION
**Pull request by sagacity**
Enables tracing in the guest (WASM) code. Doesn't work as expected yet on wasmer since we don't persist the instance across method calls.

### CODE REVIEW

**Review by gagbo — APPROVED** [review]

File: examples/example-plugin/src/lib.rs [hunk 1]
```diff
@@ -31,6 +34,8 @@ fn export_primitive_bool(arg: bool) -> bool {
 
 #[fp_export_impl(example_bindings)]
 fn export_primitive_f32(arg: f32) -> f32 {
+    tracing::info!("HELLO FROM export_primitive_f32 info");
+    log("HELLO FROM export_primitive_f32 VIA LOG".to_string());
```
Comment (arendjr) [inline examples/example-plugin/src/lib.rs:38]: This should probably be cleaned up.

File: examples/example-deno-runtime/tests.ts [hunk 2]
```diff
@@ -267,8 +267,7 @@ const imports: Imports = {
   },
 
   log: (message: string): void => {
-    // The plugin is not expected to log anything unless it panics:
-    fail("Plugin panic: " + message);
+    console.log(message);
```
Comment (arendjr) [inline examples/example-deno-runtime/tests.ts:270]: Should we have some prefix here, so it at least becomes clear where the message is coming from?

File: examples/example-plugin/src/tracing_subscriber.rs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        struct Visitor {
+            message: Option<String>,
+        }
+        impl field::Visit for Visitor {
+            fn record_debug(&mut self, field: &field::Field, value: &dyn std::fmt::Debug) {
+                if field.name() == "message" {
+                    self.message = Some(format!("{:?}", value));
+                }
+            }
+        }
```
Comment (arendjr) [inline examples/example-plugin/src/tracing_subscriber.rs:47]: I think this `struct` can be moved to the outer scope, right? I think that would make it a bit clearer.

File: examples/example-plugin/src/lib.rs (hunk 1 above)
Comment (sagacity) [reply examples/example-plugin/src/lib.rs:38]: Doh, agreed.

File: examples/example-deno-runtime/tests.ts (hunk 2 above)
Comment (sagacity) [reply examples/example-deno-runtime/tests.ts:270]: Yes, that makes sense.

**Pull request merged**